x/tokenomics/keeper/settle_session_accounting_test.go

```diff
@@ -0,0 +1,273 @@
+	// Assert that `suppliertypes.Address` account balance has *increased* by the appropriate amount
+	// Assert that `applicationAddress` account staked balance has gone to zero
+	// Assert on whatever logic we have for slashing the application or other
+}
+
+func TestSettleSessionAccounting_AppNotFound(t *testing.T) {
+	keeper, ctx, _, supplierAddr := testkeeper.TokenomicsKeeper(t)
+	wctx := sdk.WrapSDKContext(ctx)
+
+	// The base claim whose root will be customized for testing purposes
+	claim := suppliertypes.Claim{
+		SupplierAddress: supplierAddr,
+		SessionHeader: &sessiontypes.SessionHeader{
+			ApplicationAddress: sample.AccAddress(), // Random address
+			Service: &sharedtypes.Service{
+				Id:   "svc1",
+				Name: "svcName1",
+			},
+			SessionStartBlockHeight: 1,
+			SessionId:               "1",
+			SessionEndBlockHeight:   5,
+		},
+		RootHash: smstRootWithSum(42),
+	}
+
+	err := keeper.SettleSessionAccounting(wctx, &claim)
+	require.Error(t, err)
+	require.ErrorIs(t, err, types.ErrTokenomicsApplicationNotFound)
+}
+
+func TestSettleSessionAccounting_InvalidRoot(t *testing.T) {
+	keeper, ctx, appAddr, supplierAddr := testkeeper.TokenomicsKeeper(t)
+	wctx := sdk.WrapSDKContext(ctx)
+
+	// Define test cases
+	testCases := []struct {
+		desc        string
+		root        []byte // smst.MerkleRoot
+		errExpected bool
+	}{
+		{
+			desc:        "Nil Root",
+			root:        nil,
+			errExpected: true,
+		},
+		{
+			desc:        "Less than 40 bytes",
+			root:        []byte("less than 40 bytes"),
+			errExpected: true,
+		},
+		// TODO_IN_THIS_PR_DISCUSS: Should this error? Should it be on the SMT side?
+		// {
+		// 	desc:      "More than 40 bytes",
+		// 	root:      []byte("more than 40 bytes, meaning this is way too long"),
+		// 	errExpected: true,
+		// },
+		// TODO_IN_THIS_PR_DISCUSS: Should this error? Should it be on the SMT side?
+		// {
+		// 	desc: "40 bytes but empty",
+		// 	root: func() []byte {
+		// 		root := [40]byte{}
+		// 		return root[:]
+		// 	}(),
+		// 	errExpected: true,
+		// },
+		{
+			desc: "40 bytes but has an invalid value",
+			root: func() []byte {
+				var root [40]byte
+				copy(root[:], []byte("This text is exactly 40 characters!!!!!!"))
+				return root[:]
+			}(),
+			errExpected: true,
+		},
+		{
+			desc: "40 bytes and has a valid value",
+			root: func() []byte {
+				root := smstRootWithSum(42)
+				return root[:]
+			}(),
+			errExpected: false,
+		},
+	}
+
+	// Iterate over each test case
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			// Use defer-recover to catch any panic
+			defer func() {
+				if r := recover(); r != nil {
+					t.Errorf("Test panicked: %s", r)
+				}
+			}()
+
+			// Setup claim by copying the baseClaim and updating the root
+			claim := baseClaim(appAddr, supplierAddr, 0)
+			claim.RootHash = smt.MerkleRoot(tc.root[:])
```

Comment:
Should this not be:
```suggestion
			claim.RootHash = smt.MerkleRoot(tc.root())
```

Reply:
No. `root        []byte // smst.MerkleRoot`

I call the function during the test prep so it's `root` not `rootSetup`. I think its clearer.